File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        long accumulator = 1;
        int zeros = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) zeros++;
            else accumulator *= nums[i];
        }
        int[] ans = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0 && zeros == 1) ans[i] = (int) accumulator;
            else if (zeros == 0) ans[i] = (int) (accumulator / nums[i]);
            else ans[i] = 0;
        }
        return ans;
    }
}  

File: Data Structures & Algorithms/find-first-and-last-position-of-element-in-sorted-array/submission-0.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        def bSearch(nums: List[int], l: int, r: int, target: int) -> int:
            ans = -1;
            while l <= r:
                m = (l + r) // 2
                if nums[m] <= target: 
                    l = m + 1
                    ans = m
                else:
                    r = m - 1
                
            return ans

        def bSearch2(nums: List[int], l: int, r: int, target: int) -> int:
            ans = -1;
            while l <= r:
                m = (l + r) // 2
                if nums[m] < target:
                    l = m + 1
                else:
                    r = m - 1
                    ans = m
                
            return ans
        right = bSearch(nums, 0, len(nums) - 1, target)
        left = bSearch2(nums, 0, len(nums) - 1, target)

        if left == -1 or nums[left] != target:
            return [-1, -1]
        return [left, right]